### PR TITLE
feat: Orderbook

### LINKED ISSUE #14: Orderbook parsing
As discussed in [Call 2](https://github.com/joinmarket-webui/joinmarket-webui/wiki/community-call-%232), it would be interesting to do some automatic parsing of the orderbook to let the user choose sane defaults instead of fiddling with various fee settings. 

Not sure if it's feasible or easily possible, so consider this an idea more than a feature that has to be implemented. 

Listen to the call recording and have a look at the [JM orderbook docs](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/orderbook.md) and an [online orderbook](https://nixbitcoin.org/orderbook/) to get an idea of what data is in the orderbook and what settings could be automatically set or hidden from the user.

### PR DESCRIPTION
**Pull request by theborakompanioni**
Closes #14. Addresses #412.

This PR introduces basic functionality for displaying the orderbook inside Jam.

Hidden behind a feature flag for now. 

**Attention:** Also needs adaptions in the Docker image. See https://github.com/joinmarket-webui/joinmarket-webui-docker/pull/49.

Also, **important**: This feature introduces functionality that will **not** be available for users running jm "natively". The reverse proxy **is a mandatory prerequisite** for this to work as expected. If jam is not served from the reverse proxy, a user will see a generic message `Error while loading the orderbook. Your current local setup might not support fetching the orderbook. Reason: <potential reason>`.

The functionality can be further improved by eventually providing following features:
- sorting (done in #434)
- searching/filtering (done in #434)
- summary

These features are explicitely _not_ included is this PR, as setting up the orderbook and making it work is already tough to review. However, expect more to come in follow-up PRs!

image

Local (slightly outdated):
image

More Data:

This can be tested in the regtest environment without further changes, as the primary JM container exposes the orderbook on port `62601`. In order to test the "production" changes to the docker image, please follow the steps described in https://github.com/joinmarket-webui/joinmarket-webui-docker/pull/49.

### CODE REVIEW

**Comment by dnlggr** [comment]
I regularly have issue where the orderbook doesn't seem to be served on port `62601`. Lately, it feels like it's _not_ working more often than it is working. I just get an empty reply from the server on port `62601`. Any ideas what that could be? Can't find anything in the logs either.

**Comment by theborakompanioni** [comment]
> I regularly have issue where the orderbook doesn't seem to be served on port `62601`. Lately, it feels like it's _not_ working more often than it is working. I just get an empty reply from the server on port `62601`. Any ideas what that could be? Can't find anything in the logs either.

What exactly do you mean by "empty" response? Like.. no content at all? Or "just" no orders, ie. an html page including
```html
<h1>JoinMarket Orderbook</h1>
<h2>0 orders found by 0 counterparties</h2>
```
?

Locally, I experience major delays sometimes before the ob-watcher starts listening on `62601`. It can take up to several minutes (here ~5m):
```sh
> docker exec -it jm_regtest_joinmarket tail -f /root/.joinmarket/logs/obwatch_stdout.log -n 200
2022-07-26 03:17:41,153 [DEBUG]  rpc: getblockchaininfo []
[...]
2022-07-26 09:18:54,506 [DEBUG]  >>pubmsg !orderbook

started http server, visit http://0.0.0.0:62601/

2022-07-26 03:23:05,268 [DEBUG]  J57e6BFPSgkZfiq2 has dusty minsize, capping at 27300
```
(The "started http server" message has no timestamp, but is logged at the same time as the next message at 03:23:05)

If you got an empty orderbook while the secondary container is already running the maker service, it might be that JM fails to publish the offer (and subsequently fails to republish e.g. if it reconnects to irc servers). This happens in my local setup from time to time and I cannot really pinpoint to the actual problem. Anyway.. the ob-watcher should at least serve the html page.

Let me know if it is one or the other and we'll try to fix it together.

**Comment by dnlggr** [comment]
> What exactly do you mean by "empty" response? Like.. no content at all? Or "just" no orders,

Yep no content at all and then it resets the connection:

```
$ curl -v http://localhost:62601
*   Trying 127.0.0.1:62601...
* Connected to localhost (127.0.0.1) port 62601 (#0)
> GET / HTTP/1.1
> Host: localhost:62601
> User-Agent: curl/7.79.1
> Accept: */*
>
* Empty reply from server
* Closing connection 0
curl: (52) Empty reply from server
```

**Comment by theborakompanioni** [comment]
> Yep no content at all and then it resets the connection:
> 
> ```
> curl: (52) Empty reply from server
> ```

Could not reproduce by now.. hmm.. :thinking: 

Follow-up PR: #434

File: src/components/Earn.jsx [hunk 1]
```diff
@@ -99,6 +101,7 @@ export default function Earn() {
   const [isWaitingMakerStart, setIsWaitingMakerStart] = useState(false)
   const [isWaitingMakerStop, setIsWaitingMakerStop] = useState(false)
   const [isShowReport, setIsShowReport] = useState(false)
+  const [isShowOrderbook, setIsShowOrderbook] = useState(false)
```
Comment (dnlggr) [inline src/components/Earn.jsx:104]: That naming hurts a little but is consistent with what's already there, so 👍 😅

**Review by dnlggr — APPROVED** [review]

**Comment by dnlggr** [comment]
Not sure what's wrong with my setup but if it works, the orderbook parsing works fine. 👌

**Comment by dnlggr** [comment]
One improvement suggestion: Put the whole table into a [container](https://github.com/joinmarket-webui/jam/pull/430/files#diff-4188985625ac7e2a70705d02009640250956d4ae8254f03648823df1dd3c68adR376) to not have it strecht from edge to edge which looks a bit weird and is inconsistent with the rest of the app. But this can also be done in one of the upcoming PRs e.g. the one that introduces sorting etc.

**Comment by theborakompanioni** [comment]
> Not sure what's wrong with my setup but if it works, the orderbook parsing works fine. ok_hand

I am reluctant to merge it when it cannot be verified by other devs..
Ping @joinmarket-webui/implementation-contributors or @joinmarket-webui/concept-and-design-contributors : Is anybody willing to try it and verify that everything works on your end?

> One improvement suggestion: Put the whole table into a container [...]. But this can also be done in one of the upcoming PRs e.g. the one that introduces sorting etc.

Yep. Good point. Will do that in #434!

@dnlggr Would you also take a look at the docker image changes in https://github.com/joinmarket-webui/jam-docker/pull/49? Highly appreciated.

**Comment by dnlggr** [comment]
It does work sometimes so I was able to fully test it. And when it doesn't work, it's definitely not related to your changes since it's not working at the API level -- not on the UI level.

**Pull request merged**